src/test/test-sleep.c

```diff
@@ -117,6 +117,32 @@ TEST(sleep) {
         log_info("Suspend-then-Hibernate configured and possible: %s", r >= 0 ? yes_no(r) : STRERROR(r));
 }
 
+TEST(fetch_batteries_capacity_by_name) {
+        _cleanup_hashmap_free_ Hashmap *capacity = NULL;
+        int r;
+
+        assert_se(fetch_batteries_capacity_by_name(&capacity) >= 0);
+        log_debug("fetch_batteries_capacity_by_name: %u entries", hashmap_size(capacity));
+
+        const char *name;
+        void *cap;
+        HASHMAP_FOREACH_KEY(cap, name, capacity) {
+                assert(cap);  /* Anything non-null is fine. */
+                log_info("Battery %s: capacity = %i", name, get_capacity_by_name(capacity, name));
+        }
+
+        for (int i = 0; i < 5; i++) {
+                usec_t interval;
+
+                sleep(1);
```

Comment:
Why the sleep? Doing this in unit tests isn't great, since it consumes a job that can't go to other unit tests in the mean time

Reply:
We execute tests in parallel. So this doesn't really causes a slowdown of at most 5 s / number of cpus, i.e. something like 0.5 s on most laptops…

I added the sleep to exercise the code with a chance to have changing battery values. Without the sleep the result will always be 0, but with a few seconds, in principle we could get something.

Reply:
I reduced the loop to just one sleep now.